File: APCS202101_飛黃騰達(50%).cpp
#include <bits/stdc++.h>

using namespace std;

#define N 50010
typedef long long ll;

int main()
{
    // freopen("P_3_8_5.in", "r", stdin);
    int n;
    cin >> n;
    vector<pair<int, int>> point(n);
    for (int i = 0; i < n; ++i)
        cin >> point[i].first >> point[i].second;
    sort(point.begin(), point.end());
    vector<int> dp(n, 1);
    for (int i = 0; i < n; ++i)
    {
        for (int j = i + 1; j < n; ++j)
            if (point[i].second <= point[j].second)
                dp[j] = max(dp[j], dp[i] + 1);
    }
    cout << *max_element(dp.begin(), dp.end());
    return 0;
}

File: APCS202101_飛黃騰達(50%).py
n = int(input())
p = []
dp = [1]*n
dp[0] = 1
for _ in range(n):
    p.append(tuple(int(t) for t in input().split()))
p.sort()
for i in range(0, n):
    for j in range(i+1, n):
        if (p[j][1] >= p[i][1]):
            dp[j] = max(dp[j], dp[i]+1)
print(max(dp))
